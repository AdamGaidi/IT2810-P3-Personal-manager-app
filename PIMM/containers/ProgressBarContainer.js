import React, { Component } from "react";
import { ScrollView, StyleSheet, View, Text, Platform } from "react-native";
import { Notifications } from "expo";

import ProgressBar from "../components/ProgressBar";
import TextIcon from "../components/TextIcon";
import Colors from "../constants/Colors";

export default class ProgressBarContainer extends Component {
  reachedStepGoal = () => {
    const localNotification = {
      title: "Good job! You reached your step goal!",
      body: "Now you can relax, or maybe you want to walk more?",
      ios: { sound: true },
      android: {
        channelId: "steps",
        color: "yellow"
      }
    };
    Notifications.presentLocalNotificationAsync(localNotification);
  };

  goalReached() {
    this.reachedStepGoal();
    return this.props.stepsToday >= this.props.dailyGoal;
  }

  componentDidMount() {
    this.reachedStepGoal();
  }
  render() {
    const { dailyGoal, stepsToday } = this.props;

    return (
      <View style={styles.container}>
        <Text style={styles.stepBarText}>Daily steps</Text>
        <View style={styles.progressContainer}>
          <TextIcon
            defaultColor={Colors.stepBarWalkIcon}
            name={Platform.OS === "ios" ? `ios-walk` : "md-walk"}
          />
          <ProgressBar
            stepsToday={stepsToday}
            dailyGoal={dailyGoal}
            progress={stepsToday / dailyGoal}
            duration={500}
          />
          <TextIcon
            focused={this.goalReached()}
            focusedColor={Colors.stepBarFinished}
            name={
              Platform.OS === "ios"
                ? `ios-checkmark-circle${this.goalReached() ? "" : "-outline"}`
                : `md-checkmark-circle-outline`
            }
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10
  },
  progressContainer: {
    flexDirection: "row",
    alignItems: "center"
  },
  stepBarText: {
    color: Colors.stepBarText,
    marginLeft: 25
  }
});
