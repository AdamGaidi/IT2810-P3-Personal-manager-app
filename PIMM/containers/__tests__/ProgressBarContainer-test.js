import React from "react";
import ProgressBarContainer from "../ProgressBarContainer";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders shallow correctly with example props", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ProgressBarContainer stepsToday={5000} dailyGoal={10000} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
