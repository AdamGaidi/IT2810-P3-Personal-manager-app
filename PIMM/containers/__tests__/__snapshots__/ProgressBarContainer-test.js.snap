// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shallow correctly with example props 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#707070",
        "marginLeft": 25,
      }
    }
  >
    Daily steps
  </Text>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextIcon
      defaultColor="#8D8D8D"
      focusedColor="#5a95ff"
      iconSize={32}
      name="ios-walk"
    />
    <ProgressBar
      borderColor="#bebebe"
      borderRadius={10}
      borderWidth={2}
      dailyGoal={10000}
      duration={500}
      fillColor="#bebebe"
      height={18}
      progress={0.5}
      stepsToday={5000}
    />
    <TextIcon
      defaultColor="#bebebe"
      focused={false}
      focusedColor="#5a95ff"
      iconSize={32}
      name="ios-checkmark-circle-outline"
    />
  </Component>
</Component>
`;
